lib/linter.js

```diff
@@ -1007,6 +1007,35 @@ class Linter {
     }
 
     /**
+     * Verifies the text against the rules specified by the second argument.
+     * @param {string|SourceCode} textOrSourceCode The text to parse or a SourceCode object.
+     * @param {ESLintConfig} config An ESLintConfig instance to configure everything.
+     * @param {(string|Object)} [filenameOrOptions] The optional filename of the file being checked.
+     *      If this is not set, the filename will default to '<input>' in the rule context. If
+     *      an object, then it has "filename", "saveState", and "allowInlineConfig" properties.
+     * @param {boolean} [saveState] Indicates if the state from the last run should be saved.
+     *      Mostly useful for testing purposes.
+     * @param {boolean} [filenameOrOptions.allowInlineConfig] Allow/disallow inline comments' ability to change config once it is set. Defaults to true if not supplied.
+     *      Useful if you want to validate JS without comments overriding rules.
+     * @param {function(string): string[]} [filenameOrOptions.preprocess] preprocessor for source text. If provided, this should
+     *      this should accept a string of source text, and return an array of code blocks to lint.
+     * @param {function(Array<Object[]>): Object[]} [filenameOrOptions.postprocess] postprocessor for report messages. If provided,
+     *      this should accept an array of the message lists for each code block returned from the preprocessor,
+     *      apply a mapping to the messages as appropriate, and return a one-dimensional array of messages
+     * @returns {Object[]} The results as an array of messages or null if no messages.
+     */
+    verify(textOrSourceCode, config, filenameOrOptions, saveState) {
+        const preprocess = filenameOrOptions && filenameOrOptions.preprocess || (rawText => [rawText]);
+        const postprocess = filenameOrOptions && filenameOrOptions.postprocess || lodash.flatten;
+
+        return postprocess(
```

Comment:
I liked it a lot more when processors were limited to cli-engine, honestly, but I think that's a personal preference.

Reply:
I share @ilyavolodin's preference here - organizationally, it feels like `Linter` shouldn't know about processors. I don't have a specific rule of thumb or design pattern or anything like that to support that feeling, unfortunately - maybe separation of concerns, but one could argue that's too broad.

However, https://github.com/eslint/eslint/pull/9090#issuecomment-326821052 clarified for me that `postprocess` needs to be run *in between* getting errors in `verify` and applying fixes. I don't see an easy way around that, so thankfully `preprocess` and `postprocess`can be treated as black boxes from `Linter`'s perspective.